app/controllers/api/pictures_controller.rb

```diff
@@ -1,11 +1,26 @@
 module Api
   class PicturesController < BaseController
     def create_resource(_type, _id, data)
+      validate_picture_data(data)
```

Comment:
@jntullo I'm guessing that if there was (or ever is) a way to great a bogus picture outside of this controller, and that's not wanted for the same reason, then it may belong in the model. I am assuming here that this is the case, but I don't know if that is so. But if it is, is there a way to push what logic you have here down into the model and then maybe iterate on it to see if you can delegate to Paperclip as a follow up?